File: vignerecipher/vigenereCipher.cpp
#include <iostream>
#include <string>

std::string vigenereCipherEncrypt(const std::string& plaintext, const std::string& key) {
    std::string encryptedText;
    int keyLength = key.length();
    
    for (int i = 0; i < plaintext.length(); i++) {
        char plainChar = plaintext[i];
        char keyChar = key[i % keyLength];
        
        if (isalpha(plainChar)) {
            char base = isupper(plainChar) ? 'A' : 'a';
            char encryptedChar = ((plainChar - base + keyChar - 'A') % 26) + base;
            encryptedText += encryptedChar;
        } else {
            encryptedText += plainChar;  // Keep non-alphabet characters as is.
        }
    }
    
    return encryptedText;
}

int main() {
    std::string plaintext;
    std::string key;
    
    std::cout << "Enter the plaintext: ";
    std::getline(std::cin, plaintext);
    
    std::cout << "Enter the Vigenere key: ";
    std::cin >> key;

    std::string encryptedText = vigenereCipherEncrypt(plaintext, key);
    std::cout << "Encrypted: " << encryptedText << std::endl;

    return 0;
}

File: vignerecipher/VigenereCipher.py
def vigenere_cipher_encrypt(plaintext, key):
    encrypted_text = []
    key_length = len(key)
    
    for i, char in enumerate(plaintext):
        if char.isalpha():
            key_char = key[i % key_length]
            shift = ord(key_char) - ord('A' if key_char.isupper() else 'a')
            if char.isupper():
                encrypted_char = chr((ord(char) - ord('A') + shift) % 26 + ord('A'))
            else:
                encrypted_char = chr((ord(char) - ord('a') + shift) % 26 + ord('a'))
            encrypted_text.append(encrypted_char)
        else:
            encrypted_text.append(char)  # Keep non-alphabet characters as is.
    
    return ''.join(encrypted_text)

plaintext = input("Enter the plaintext: ")
key = input("Enter the Vigenere key: ")

encrypted_text = vigenere_cipher_encrypt(plaintext, key)
print("Encrypted:", encrypted_text)
